test/jb_splitsStore/set.test.js

```diff
@@ -0,0 +1,293 @@
+    let jbSplitsStore = await jbSplitsStoreFact.deploy(
+      /*operator=*/mockOperatorStore.address,
+      /*projects=*/mockProjects.address,
+      /*directory=*/mockDirectory.address
+    );
+
+    let splits = createSplitArray(addrs[0].address, 4);
+
+    return { deployer, projectOwner, addrs, jbSplitsStore, splits, mockOperatorStore, mockProjects, mockDirectory };
+  }
+
+  // Create array of JBSplit struct
+  function createSplitArray(_benefAdress, n) {
+    let splits = []
+    for (let i = 0; i < n; i++) {
+      splits.push({
+        preferClaimed: false,
+        percent: Math.floor(10000000 / n),
+        lockedUntil: 0,
+        beneficiary: _benefAdress,
+        allocator: ZERO_ADDRESS,
+        projectId: 0
+      });
+    }
+    return splits;
+  }
+
+  // Expected behavior when no preexisting splits for this group - happy path
+  it('set() and corresponding events', async function () {
+
+    const { projectOwner, addrs, jbSplitsStore, splits, mockOperatorStore, mockDirectory } = await setup();
+
+    // set(..) called from project owner address (which is set in setup() )
+    await mockOperatorStore.mock.hasPermission.returns(false);
+    await mockDirectory.mock.controllerOf.returns(addrs[0].address);
+
+    const tx = await jbSplitsStore.connect(projectOwner).set(
+      /*projectId=*/PROJECT_ID,
+      /*domain=*/DOMAIN,
+      /*group=*/GROUP,
+      /*splits[]=*/splits
+    );
+
+    // Expect one event per split in splits[]
+    await expect(tx)
+      .to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+      .and.to.emit(jbSplitsStore, 'SetSplit')
+
+    // Get the current splits (for this proj/dom/group)
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    for (let idx = 0; idx < splits.length; idx++) {
+      for (let split_key of Object.keys(splits[idx])) {
+        expect(splitsStored[idx].split_key).to.equal(splits[idx].split_key);
+      }
+    }
+  })
+
+  // New splits[] when preexisting ones already exist without any lockedUntil
+  it('New splits', async function () {
+    const { projectOwner, addrs, jbSplitsStore, splits } = await setup();
+
+    await jbSplitsStore.connect(projectOwner).set(
+      /*projectId=*/PROJECT_ID,
+      /*domain=*/DOMAIN,
+      /*group=*/GROUP,
+      /*splits[]=*/splits
+    );
+
+    // 4 new ones, with a new beneficiary for each,
+    let newBeneficiary = addrs[5].address;
+    let newSplits = createSplitArray(newBeneficiary, 4);
+
+    await jbSplitsStore.connect(projectOwner).set(
+      /*projectId=*/PROJECT_ID,
+      /*domain=*/DOMAIN,
+      /*group=*/GROUP,
+      /*splits[]=*/newSplits
+    );
+
+    // Get the splits[] curently stored
+    let splitsStored = await jbSplitsStore.splitsOf(PROJECT_ID, DOMAIN, GROUP);
+
+    for (let idx = 0; idx < splits.length; idx++) {
+      for (let split_key of Object.keys(newSplits[idx])) {
```

Comment:
I feel in this test and in others the following would be cleaner:

for (let split of newSplits) {
  for( let key of Object.keys(split)) {
    ...
  }
}

Replace here and everywhere else.